### PR TITLE
QOL-8842 test cleanup

### PR DESCRIPTION
**Pull request by ThrawnCA**
- Use fixed CKAN core version in tests
- Update ckanext-qgov test version
- Reduce dependency on in-browser test setup
- Point unnecessary third-party domains at localhost during tests for faster resolution

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -91,6 +91,10 @@ services:
     networks:
       - amazeeio-network
       - default
+    extra_hosts:
+      - "www.googletagmanager.com:127.0.0.1"
```
Comment (duttonw) [inline docker-compose.yml:95]: can you add comment why we added these for future us.

File: test/features/login_redirect.feature [hunk 2]
```diff
@@ -22,23 +22,15 @@ Feature: Login Redirection
         When I log in directly
         Then I should see "Change details"
 
-    @dataset_setup
-    Scenario: As a Sysadmin I set the visibility of a public record to private for the following scenarios
```
Comment (duttonw) [inline test/features/login_redirect.feature:26]: very good pickup on not changing data mid way through (unless we create a new one during test background)

File: test/features/steps/steps.py [hunk 3]
```diff
@@ -213,10 +213,15 @@ def create_dataset(context, license, file_format, file):
...
 
-# The default behaving step does not convert base64 emails
-# Modified the default step to decode the payload from base64
 @step(u'I should receive a base64 email at "{address}" containing "{text}"')
 def should_receive_base64_email_containing_text(context, address, text):
+    should_receive_base64_email_containing_texts(context, address, text, None)
+
+
+@step(u'I should receive a base64 email at "{address}" containing both "{text}" and "{text2}"')
+def should_receive_base64_email_containing_texts(context, address, text, text2):
```
Comment (duttonw) [inline test/features/steps/steps.py:222]: nice way to ensure we don't get race conditions on doing a double check.
you could also do 'subject' contains text also.

File: .docker/scripts/create-test-data.sh [hunk 4]
```diff
@@ -97,7 +97,7 @@ add_user_if_needed walker "Walker" walker@localhost
 
 # Create test dataset with our standard fields
 curl -LsH "Authorization: ${API_KEY}" \
-    --data '{"name": "test-dataset", "owner_org": "'"${TEST_ORG_ID}"'",
+    --data '{"name": "test-dataset", "owner_org": "'"${TEST_ORG_ID}"'", "private": true,
```
Comment (duttonw) [inline .docker/scripts/create-test-data.sh:100]: do we need another one for public public?

**Review by duttonw — APPROVED** [review]

File: .docker/scripts/create-test-data.sh (hunk 4 above)
Comment (ThrawnCA) [reply .docker/scripts/create-test-data.sh:100]: We can do that if we decide a test needs it. The sample data like Anna Karenina generally handles public access though.

**Pull request merged**